templates/routing-legacy/main.go

```diff
@@ -0,0 +1,59 @@
+// package main holds the implementation of the nextroute template.
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/nextmv-io/sdk/nextroute"
+	"github.com/nextmv-io/sdk/nextroute/factory"
+	"github.com/nextmv-io/sdk/nextroute/schema"
+	"github.com/nextmv-io/sdk/run"
+	runSchema "github.com/nextmv-io/sdk/run/schema"
+)
+
+func main() {
+	runner := run.CLI(solver)
+	err := runner.Run(context.Background())
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type options struct {
+	Model  factory.Options                `json:"model,omitempty"`
+	Solve  nextroute.ParallelSolveOptions `json:"solve,omitempty"`
+	Format nextroute.FormatOptions        `json:"format,omitempty"`
+}
+
+func solver(
+	ctx context.Context,
+	input schema.FleetInput,
+	options options,
+) (runSchema.Output, error) {
+	nextrouteInput, err := schema.FleetToNextRoute(input)
+	if err != nil {
+		panic(err)
+	}
+
+	model, err := factory.NewModel(nextrouteInput, options.Model)
+	if err != nil {
+		return runSchema.Output{}, err
+	}
+
+	solver, err := nextroute.NewParallelSolver(model)
+	if err != nil {
+		return runSchema.Output{}, err
+	}
+
+	solutions, err := solver.Solve(ctx, options.Solve)
+	if err != nil {
+		return runSchema.Output{}, err
+	}
+	last := solutions.Last()
+
+	output := factory.Format(ctx, options, solver, last)
```

Comment:
We need to make this also backwards compatible. So we need a new format function as well that transforms to legacy output.

Reply:
Yep, good catch.

Reply:
legacy output will be a follow up. Ticket is here: https://linear.app/nextmv/issue/ENG-3702/add-legacy-output-to-nextroute

Reply:
Ok.